contracts/Compass.vy

```diff
@@ -39,38 +41,45 @@ struct TokenSendArgs:
 event ValsetUpdated:
     checkpoint: bytes32
     valset_id: uint256
+    event_id: uint256
 
 event LogicCallEvent:
     logic_contract_address: address
     payload: Bytes[MAX_PAYLOAD]
     message_id: uint256
+    event_id: uint256
 
 event SendToPalomaEvent:
     token: address
     sender: address
     receiver: String[64]
     amount: uint256
+    event_id: uint256
 
 event BatchSendEvent:
     token: address
-    message_id: uint256
+    batch_id: uint256
+    event_id: uint256
 
 event ERC20DeployedEvent:
     paloma_denom: String[64]
     token_contract: address
     name: String[64]
     symbol: String[32]
     decimals: uint8
+    event_id: uint256
 
 last_checkpoint: public(bytes32)
 last_valset_id: public(uint256)
+last_event_id: public(uint256)
+last_batch_id: public(HashMap[address, uint256])
 message_id_used: public(HashMap[uint256, bool])
 
 # compass_id: unique identifier for compass instance
 # valset: initial validator set
 @external
-def __init__(compass_id: bytes32, valset: Valset):
-    COMPASS_ID = compass_id
+def __init__(_compass_id: bytes32, _event_id: uint256, valset: Valset):
+    compass_id = _compass_id
```

Comment:
Do you not need to initialise the event ID with the constructor arguments as well?